src/mpl.cpp

```diff
@@ -5,187 +5,194 @@
-    - Perform WHOAMI check
-    - Trigger reset and wait for completion
-    - Set altimeter mode
-    - Set oversampling ratio to 2^128 (OS128)
-    - Enable data ready events for pressure/altitude and temperature
-*/
-hal::status mpl::driver_configure()
-{
-    // sanity check
-    std::array<hal::byte, 1> whoami_payload { WHOAMI_R };
-    std::array<hal::byte, 1> whoami_buffer {};
-    HAL_CHECK(hal::write_then_read(*m_i2c, device_address, whoami_payload, whoami_buffer, hal::never_timeout()));
+/***********************************************
+ *  Free Functions
+ ***********************************************/
 
-    if (whoami_buffer[0] != 0xC4) {
-        return hal::new_error();
-    }
-
-    // software reset
-    modify_reg_bits(CTRL_REG1, CTRL_REG1_RST);
-
-    poll_flag(CTRL_REG1, CTRL_REG1_RST, false);
-
-    // set oversampling ratio to 2^128 and set altitude mode
-    modify_reg_bits(CTRL_REG1, CTRL_REG1_OS128 | CTRL_REG1_ALT );
-    sensor_mode = ALTIMETER_M;
-
-    // enable data ready events for pressure/altitude and temperature
-    std::array<hal::byte, 2> dr_payload {
-        PT_DATA_CFG_R, 
-        PT_DATA_CFG_TDEFE |
-        PT_DATA_CFG_PDEFE |
-        PT_DATA_CFG_DREM
-    };
-    HAL_CHECK(hal::write(*m_i2c, device_address, dr_payload, hal::never_timeout()));
-
-    return hal::success();
-}
-
-/*
-* @brief Set the CTRL_REG1_ALT bit in CTRL_REG1 to the value corresponding to 'mode'
+/**
+* @brief Set the ctrl_reg1_alt bit in ctrl_reg1 to the value corresponding to 'mode'
+* @param p_i2c The I2C peripheral used for communication with the device.
 * @param mode: The desired operation mode
 */
-hal::status mpl::set_mode(mpl_mode_t mode)
+hal::status set_mode(hal::i2c* p_i2c, mpl::mpl_mode_t mode)
 {
-    // Read value of CTRL_REG1
-    std::array<hal::byte, 1> ctrl_payload { CTRL_REG1 };
-    std::array<hal::byte, 1> ctrl_buffer;
-    HAL_CHECK(hal::write_then_read(*m_i2c, device_address, ctrl_payload, ctrl_buffer, hal::never_timeout()));
-
-    // Set mode ctrl reg bit to binary value of 'mode'
-    hal::byte mode_set_reg = ctrl_buffer[0] ^= (-mode ^ ctrl_buffer[0]) & CTRL_REG1_ALT;
-    std::array<hal::byte, 2> ctrl_write { CTRL_REG1, mode_set_reg };
-    HAL_CHECK(hal::write(*m_i2c, device_address, ctrl_write, hal::never_timeout()));
-
-    sensor_mode = mode;
-
-    return hal::success();
-}
-
-/*
-* @brief Trigger one-shot measurement by setting 
-    CTRL_REG1_OST bit in CTRL_REG1.
-*/
-hal::status mpl::initiate_one_shot() 
-{
-    // Wait for one-shot flag to clear
-    poll_flag(CTRL_REG1, CTRL_REG1_OST, false);
+    // Read value of ctrl_reg1
+    auto ctrl_buffer = HAL_CHECK(hal::write_then_read<1>(*p_i2c, device_address, std::array<hal::byte, 1>{ ctrl_reg1 }, hal::never_timeout()));
+    
+    hal::byte reg_val = ctrl_buffer[0];
+    if ((reg_val & ctrl_reg1_alt) != static_cast<int>(mode)) {
+        // Set mode ctrl reg bit to binary value of 'mode'
+        if (mode == mpl::mpl_mode_t::BAROMETER_M) {
```

Comment:
```suggestion
        if (mode == mpl::mpl_mode_t::BAROMETER_M) {
```

CAP_SNAKE_CASE is reserved for macros. This should be lower case. Also what does M and A stand for? Altitude or meters? If so, we should spell that out to make the code more readable without needing additional comments to explain.